.github/workflows/licenses.yml

```diff
@@ -31,19 +31,13 @@ jobs:
           git config --global user.email "ci@ci.invalid"
           git config --global user.name "Discourse CI"
 
-      - name: Bundler cache
-        uses: actions/cache@v4
-        with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-gem-${{ hashFiles('**/Gemfile.lock') }}
-          restore-keys: ${{ runner.os }}-gem-
-
-      - name: Setup gems
+      - name: bundle install
         run: |
-          bundle config --local path vendor/bundle
+          cp -r /var/www/discourse/vendor/bundle ./vendor/bundle
+          gem install bundler --conservative -v $(awk '/BUNDLED WITH/ { getline; gsub(/ /,""); print $0 }' Gemfile.lock)
           bundle config --local deployment true
           bundle config --local without development
-          bundle install --jobs 4
+          bundle install --jobs $(($(nproc) - 1))
```

Comment:
This seems risky on a high-cpu machine. I think we should cap it at something sensible.

Reply:
Good point. I think 4 is good enough since we are not expected to be installing alot of gems most of the time.

Reply:
Why not just set it to 4 unconditionally? This is a network-bound operation, not cpu-bound. So all the `nproc` complexity seems unnecessary?

Reply:
I live for complexity 😛 I'll change it back to 4.